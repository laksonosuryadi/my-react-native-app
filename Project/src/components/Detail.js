import React, { Component } from 'react';
import {
  Text,
  View,
  Image,
  ScrollView,
} from 'react-native';

import { Icon, Col, Grid } from 'native-base';

class Detail extends Component {
  constructor(props){
    super(props)
  }

  static navigationOptions = { title : '700px' }

  render() {
    const {params} = this.props.navigation.state
    return(
      <View style={styles.content}>
        <ScrollView>
          <View style={styles.header}>
            <Image
                style={{width:60, height: 60, borderRadius: 100}}
                source={{uri: params.detail.user.userpic_url}}
            />
            <Text style={styles.headerText}>{params.detail.user.fullname}</Text>
          </View>
          <Image
              style={{width:400, height: 400}}
              source={{uri: params.detail.image_url}}
          />
          <View style={styles.description}>
            <Text style={{fontSize: 20, fontWeight:'bold', marginBottom: 10}}>{params.detail.name}</Text>
            <View style={styles.spaced}>
              <Grid>
                <Col style={{flexDirection:'row', alignItems:'center'}}>
                  <Icon name="md-heart" style={{fontSize: 30, color:'red'}}/>
                  <Text style={{marginLeft:10}}>{params.detail.votes_count}</Text>
                </Col>
                <Col style={{flexDirection:'row', alignItems:'center'}}>
                  <Icon name="md-eye" style={{fontSize: 30}}/>
                  <Text style={{marginLeft:10}}>{params.detail.times_viewed}</Text>
                </Col>
              </Grid>
            </View>
            <Text>Camera: {params.detail.camera}</Text>
            <Text>ISO: {params.detail.iso}</Text>
            <Text>Focal Length: {params.detail.focal_length}</Text>
            <Text>Aperture: {params.detail.aperture}</Text>
            <Text>Shutter Speed: {params.detail.shutter_speed}</Text>
          </View>
        </ScrollView>
      </View>
    )
  }

}

const styles = {
  content: {
    justifyContent: 'center',
    alignItems: 'center',
    width: '100%'
  },
  header: {
    flexDirection: 'row',
    alignContent: 'center',
    alignItems: 'center',
    padding: 10
  },
  headerText: {
    marginLeft: 10,
    fontWeight:'bold',
    color: '#000'
  },
  description: {
    paddingLeft: 15,
    paddingTop: 15,
    marginBottom: 20
  },
  spaced: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    marginBottom: 10
  }
}

export default Detail
