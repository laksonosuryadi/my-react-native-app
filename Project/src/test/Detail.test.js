import renderer from 'react-test-renderer';
import React from 'react';
import ReactDOM from 'react-dom';

import Detail from '../components/Detail'

it('render Detail successfully', () => {
  const tree = renderer.create(
      <Detail
        navigation={{
          state: {
            params:
            {
              detail:
              {
                user:
                {
                  userpic_url: ''
                },
                image_url: '',
                votes_count: 0,
                times_viewed: 0,
                camera: '',
                iso: '',
                focal_length: 0,
                aperture: 0,
                shutter_speed: 0
              }
            }
          }
        }}
      />
  )
  expect(tree).toMatchSnapshot();
})
