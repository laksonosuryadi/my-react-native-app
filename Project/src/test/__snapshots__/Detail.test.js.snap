// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Detail successfully 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "flexDirection": "row",
            "padding": 10,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "",
            }
          }
          style={
            Object {
              "borderRadius": 100,
              "height": 60,
              "width": 60,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#000",
              "fontWeight": "bold",
              "marginLeft": 10,
            }
          }
        />
      </View>
      <Image
        source={
          Object {
            "uri": "",
          }
        }
        style={
          Object {
            "height": 400,
            "width": 400,
          }
        }
      />
      <View
        style={
          Object {
            "marginBottom": 20,
            "paddingLeft": 15,
            "paddingTop": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 20,
              "fontWeight": "bold",
              "marginBottom": 10,
            }
          }
        />
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-around",
              "marginBottom": 10,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    Object {
                      "color": "red",
                      "fontSize": 30,
                    },
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
                virtual={undefined}
              >
                
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "marginLeft": 10,
                  }
                }
              >
                0
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    Object {
                      "color": "#000",
                      "fontSize": 30,
                    },
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
                virtual={undefined}
              >
                
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "marginLeft": 10,
                  }
                }
              >
                0
              </Text>
            </View>
          </View>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Camera: 
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          ISO: 
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Focal Length: 
          0
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Aperture: 
          0
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Shutter Speed: 
          0
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
